trunner/harness/unity.py

```diff
@@ -27,14 +27,15 @@ def unity_harness(dut: Dut, ctx: TestContext, result: TestResult) -> Optional[Te
             if parsed["status"] in ["FAIL", "IGNORE"]:
                 last_assertion = parsed
         elif idx in (1, 2):
+            msg = ""
             if last_assertion:
-                parsed["msg"] = last_assertion["msg"]
+                msg = last_assertion["msg"]
                 last_assertion = {}
 
             status = Status.from_str(parsed["status"])
             subname = f"{parsed['group']}.{parsed['name']}"
             if "path" in parsed and "line" in parsed:
-                parsed["msg"] = f"[{parsed['path']}:{parsed['line']}] " + parsed["msg"]
+                parsed["msg"] = f"[{parsed['path']}:{parsed['line']}] " + msg
```

Comment:
- this means that test can fail without any assert? 
 - please note that you're changing the semantics (the msg would be added to subresult only if `path` and `line` are present - not sure if relevant. Maybe just make a minor change (without the changes above):
```suggestion
                parsed["msg"] = f"[{parsed['path']}:{parsed['line']}] " + parsed.get("msg", "")
```

Reply:
Hmm, from what I understand now, the issue is that the test might fail with an assert, but the regex won't catch it due to the lack of message on the end of assert. In that case, I would adjust the regex, that this `: (?P<msg>.*?)` can be optional

Reply:
Test can fail without any message, for example using `TEST_FAIL()` and in this situation there was an error:
![image](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/108683715/ec19fac5-6b47-4972-a9cd-808acd1fe567)

Reply:
I'm not sure writing tests which can fail without an explicit reason should be encouraged.

Still, adding `parsed.get("msg", "")` probably saves the day?

Reply:
Yea, I agree, but I think such a case should be taken into account and handled. I used your suggestion and also adjusted assert regex.